core/amber/src/main/python/core/python_worker.py

```diff
@@ -31,12 +57,66 @@ def run(self) -> None:
         )
         main_loop_thread = Thread(target=self._main_loop.run, name="main_loop_thread")
 
+        stop_event = Event()
+        heartbeat_thread = Thread(
+            target=self.heartbeat,
+            name="heartbeat_thread",
+            args=(5.0, stop_event, network_sender_thread, main_loop_thread),
+        )
+
         network_sender_thread.start()
         main_loop_thread.start()
+        heartbeat_thread.start()
         main_loop_thread.join()
         network_sender_thread.join()
+        with self.condition:
+            self.condition.notify()
+
+        heartbeat_thread.join()
 
     @overrides
     def stop(self):
         self._main_loop.stop()
         self._network_sender.stop()
+        self_clean_child_process()
+
+    def heartbeat(self, interval, stop_event, network_sender_thread, main_loop_thread):
+        while network_sender_thread.is_alive() and main_loop_thread.is_alive():
+            try:
+                socket.create_connection(
+                    (self.parsed_server_host, self.parsed_server_port), timeout=1
```

Comment:
I am not quite understand this logic. it seems you are trying to create a new socket to the (host, port) pair. My understanding is that the flight client already has a socket connected with this (host, port) pair. Isn't that a conflict?